### ISSUE TITLE
rrdDef.getPath() Does not return correct path in case of Windows machine

### ISSUE THREAD

**Issue by sagarmiglani** [rrd4j/rrd4j#152]
If create a `RrdDef` on windows machine with the code below:
```
String path = "C:\\playground\\rrd4j\\metrics\\metrics.rrd";
RrdDef def = new RrdDef(path, 5); 
```
`def.getPath()` returns the path as `/C:/playground/rrd4j/metrics/metrics.rrd` i.e the path starts with a '/'

And If I want to create a `RrdDb` using the path returned by the `RrdDef` ( `RrdDb.of(def.getPath())` ), I get the following exception:
```
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char <:> at index 2: /C:/playground/rrd4j/metrics/metrics.rrd
	at sun.nio.fs.WindowsPathParser.normalize(WindowsPathParser.java:182)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:153)
	at sun.nio.fs.WindowsPathParser.parse(WindowsPathParser.java:77)
	at sun.nio.fs.WindowsPath.parse(WindowsPath.java:94)
	at sun.nio.fs.WindowsFileSystem.getPath(WindowsFileSystem.java:255)
	at java.nio.file.Paths.get(Paths.java:84)
	at org.rrd4j.core.RrdFileBackendFactory.exists(RrdFileBackendFactory.java:26)
	at org.rrd4j.core.RrdBackendFactory.exists(RrdBackendFactory.java:543)
	at org.rrd4j.core.RrdDb.<init>(RrdDb.java:624)
	at org.rrd4j.core.RrdDb.of(RrdDb.java:500)
```

**Comment by fbacchella** [comment]
What version are you using ? In current version (3.8.1) , the code used is:
```
new File(rrdpath).toURI();
```

Not sure what it should return on windows.

**Comment by sagarmiglani** [comment]
Using the latest version (3.8.1), `new File(rrdpath).toURI().toPath()` returns a path starting with '/' on windows

**Comment by sagarmiglani** [comment]
Not really sure, probably `RrdDef.getPath()` should use `Paths.get(this.uri).toString()` instead of `this.uri.getPath()`

**Comment by fbacchella** [comment]
The purpose is to be able to handle any URI scheme. Internally, all paths are stored as URI. For windows, the way to go is:

    RrdDb.of(def.getUri());

But for improved reliability, I just commited patch [ffd6bb2](https://github.com/rrd4j/rrd4j/commit/ffd6bb27b15b5a62317141f8d5462160d5956ef9)

**Comment by fbacchella** [comment]
A snapshot version is available at https://oss.sonatype.org/content/repositories/snapshots/org/rrd4j/rrd4j/3.8.2-SNAPSHOT/. Can you test it ? I don’t have a Windows OS to try it.

**Comment by sagarmiglani** [comment]
Thanks for the patch @fbacchella. Now the path returned for the above case seems to be correct in case of windows machine.
`RrdDef.getPath()` returns `C:\playground\rrd4j\metrics\metrics.rrd` for `C:\playground\rrd4j\metrics\metrics.rrd`

Apart from that I just wanted to highlight that in case of windows machine we now have inconsistencies in the file separator of an absolute and relative path returned by `RrdDef.getPath()`.

For a relative path, Let's say `metrics\metrics.rrd`; `RrdDef.getPath()` returns `metrics/metrics.rrd` while for absolute path we have the file separator `\` as in the case above. It is because while creating the uri of a relative path, we are replacing the file separator to `/` : https://github.com/rrd4j/rrd4j/blob/d3407d3aec52730be20a31faae8617330eafc66f/src/main/java/org/rrd4j/core/RrdBackendFactory.java#L300

**Comment by fbacchella** [comment]
It’s because, it’s stored as an URI internally. I should had a comment on RrdDef.getPath() is unreliable and one should use getUri() instead.

**Comment by sagarmiglani** [comment]
It would be helpful if we can add that comment.
Thanks @fbacchella for the help

**Comment by fbacchella** [comment]
Fixed in 3.8.2.
